src/Databases/DatabaseReplicated.h

```diff
@@ -107,6 +112,7 @@ class DatabaseReplicated : public DatabaseAtomic
     zkutil::ZooKeeperPtr getZooKeeper() const;
 
     std::atomic_bool is_readonly = true;
+    bool is_recovering = false;
```

Comment:
Shouldn't this be atomic also?
Recovery is done in a DDLWorker thread, while the check `allowMoveTableToOtherDatabaseEngine` is called on rename.